File: server/dna.core/dna.core.service/Models/UserDetailModel.cs
using dna.core.service.Models.Abstract;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace dna.core.service.Models
{
    public class UserDetailModel : IModelBase
    {
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required, MaxLength(100)]
        public string FirstName { get; set; }

        [Required, MaxLength(100)]
        public string LastName { get; set; }

        [MaxLength(100)]
        public string Avatar { get; set; }
        public virtual UserModel User { get; set; }
    }
}

File: server/medicore.apps/src/model/userDetailModel.ts
import { Type } from 'class-transformer';
import { UserModel } from './userModel';
import { IEntityBase } from './IEntityBase';

export class UserDetailModel implements IEntityBase {
    id: number;
    userId: number;
    firstName: string;
    lastName: string;

    //region model
    @Type(() => UserModel)
    user: UserModel;
}

